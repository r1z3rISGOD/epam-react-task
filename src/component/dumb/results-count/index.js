import {ResultsCount} from "./results-count"
export {ResultsCount}
