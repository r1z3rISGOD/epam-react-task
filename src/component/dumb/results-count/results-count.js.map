{"version":3,"file":"results-count.js","sourceRoot":"","sources":["results-count.tsx"],"names":[],"mappings":";;;AAAA,6BAA8B;AAC9B,gCAA6B;AAMtB,IAAM,YAAY,GAAiC,UAAC,EAAgB;QAAd,YAAY,kBAAA;IACvE,IAAI,YAAY,GAAG,CAAC,EAAE;QACpB,OAAO,CACC,8BAAM,SAAS,EAAC,eAAe;YAC9B,YAAY;4BACV,CACV,CAAA;KACF;IAED,OAAO,CACG,yCAAK,CACd,CAAA;AACH,CAAC,CAAA;AAZY,QAAA,YAAY,gBAYxB"}