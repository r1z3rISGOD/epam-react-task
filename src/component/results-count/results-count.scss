$font: "Roboto", sans-serif;


.results-count {
  font-size: 1.5rem;
  font-family: $font;
}
