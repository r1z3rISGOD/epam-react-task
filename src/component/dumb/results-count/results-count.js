"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResultsCount = void 0;
var React = require("react");
require("./results-count.scss");
var ResultsCount = function (_a) {
    var resultsCount = _a.resultsCount;
    if (resultsCount > 0) {
        return (React.createElement("span", { className: "results-count" },
            resultsCount,
            " movies found"));
    }
    else {
        return (React.createElement(React.Fragment, null));
    }
};
exports.ResultsCount = ResultsCount;
//# sourceMappingURL=results-count.js.map