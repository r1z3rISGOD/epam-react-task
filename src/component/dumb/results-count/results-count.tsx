import * as React from 'react'
import './results-count.scss'

interface ResultsCountProps {
  resultsCount: number
}

export const ResultsCount : React.FC<ResultsCountProps> = ({ resultsCount }) => {
  if (resultsCount > 0) {
    return (
            <span className="results-count">
            {resultsCount} movies found
        </span>
    )
  }

  return (
            <></>
  )
}
