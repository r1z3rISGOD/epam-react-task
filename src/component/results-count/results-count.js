import React from "react";
import "./results-count.scss"

export default function ResultsCount() {
    return(
        <span className="results-count">
            7 movies found
        </span>
    )
}
