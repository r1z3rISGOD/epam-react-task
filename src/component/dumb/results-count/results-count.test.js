import { ResultsCount } from './results-count.tsx'
import React from 'react'

describe('results count tests', () => {
  it('results count test 1', () => {
    const comp = shallow(<ResultsCount resultsCount={5}/>)
    expect(comp).toMatchSnapshot()
  })
  it('results count test 2', () => {
    const comp = shallow(<ResultsCount resultsCount={0}/>)
    expect(comp).toMatchSnapshot()
  })
})
