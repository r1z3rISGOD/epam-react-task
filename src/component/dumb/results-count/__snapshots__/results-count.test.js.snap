// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`results count tests results count test 1 1`] = `
<span
  className="results-count"
>
  5
   movies found
</span>
`;

exports[`results count tests results count test 2 1`] = `<Fragment />`;
